File: modulea/src/main/java/com/braisgabin/dagger/modulea/ComponentA.java
package com.braisgabin.dagger.modulea;


import com.braisgabin.dagger.moduleb.ComponentB;
import dagger.Component;

@Component(dependencies = ComponentB.class)
public interface ComponentA {

    @Component.Builder
    interface Builder {

        Builder with(ComponentB component);

        ComponentA build();
    }

    String key();
}

File: modulea/src/main/java/com/braisgabin/dagger/modulea/ComponentA.kt
package com.braisgabin.dagger.modulea

import com.braisgabin.dagger.moduleb.ComponentB
import dagger.Component

@Component(dependencies = [ComponentB::class])
interface ComponentA {

    @Component.Builder
    interface Builder {

        fun with(component: ComponentB): Builder

        fun build(): ComponentA
    }

    fun key(): String
}
